File: patterns/proxy-item/rules-python/proxy_item.py
from openhab import rule, Registry
from openhab.triggers import ItemCommandTrigger, ItemStateUpdateTrigger


@rule(triggers=[ItemCommandTrigger("ProxySwitch")])
class ProxySwitchErhieltBefehl:
    def execute(self, module, input):
        event = input.get("event")
        if not event:
            return

        command = str(event.getItemCommand())
        bound = Registry.getItem("BoundSwitch")
        if str(bound.getState()) != command:
            bound.sendCommand(command)
            self.logger.info("ProxySwitch -> BoundSwitch: " + command)


@rule(triggers=[ItemStateUpdateTrigger("BoundSwitchUpdates")])
class BoundSwitchUpdatesErhieltUpdate:
    def execute(self, module, input):
        event = input.get("event")
        if not event:
            return

        state = str(event.getItemState())
        proxy = Registry.getItem("ProxySwitch")
        if str(proxy.getState()) != state:
            proxy.postUpdate(state)
            self.logger.info("BoundSwitchUpdates -> ProxySwitch: " + state)

File: patterns/proxy-item/rules-js/proxy-item.js
rules.JSRule({
  name: "ProxySwitch erhielt Befehl",
  triggers: [triggers.ItemCommandTrigger("ProxySwitch")],
  execute: (event) => {
    const commandStr = String(event.receivedCommand);
    if (items.getItem("BoundSwitch").state !== commandStr) {
      items.getItem("BoundSwitch").sendCommand(event.receivedCommand);
      console.log("ProxySwitch -> BoundSwitch: " + commandStr);
    }
  }
});

rules.JSRule({
  name: "BoundSwitchUpdates erhielt Update",
  triggers: [triggers.ItemStateUpdateTrigger("BoundSwitchUpdates")],
  execute: (event) => {
    const stateStr = String(event.receivedState);
    if (items.getItem("ProxySwitch").state !== stateStr) {
      items.getItem("ProxySwitch").postUpdate(event.receivedState);
      console.log("BoundSwitchUpdates -> ProxySwitch: " + stateStr);
    }
  }
});
